controller/array_action/array_mediator_svc.py

```diff
@@ -210,7 +210,8 @@ def _generate_snapshot_response_with_verification(self, cli_object):
         fcmap = self._get_fcmap_as_target_if_exists(cli_object.name)
         if fcmap is None or fcmap.copy_rate != '0':
             raise array_errors.ExpectedSnapshotButFoundVolumeError(cli_object.name, self.endpoint)
-        return self._generate_snapshot_response(cli_object, fcmap.source_vdisk_name)
+        source_volume_id = self._get_wwn_by_volume_name_if_exists(fcmap.source_vdisk_name)
```

Comment:
why not put it inside the "generate_snapshot" function, as done in "generate_volume"?